observer/mon_conf.go

```diff
@@ -0,0 +1,63 @@
+package observer
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/letsencrypt/boulder/cmd"
+	p "github.com/letsencrypt/boulder/observer/probes"
+	"gopkg.in/yaml.v2"
+)
+
+type settings map[string]interface{}
+
+// MonConf is exported to receive yaml configuration
+type MonConf struct {
+	Valid    bool
+	Period   cmd.ConfigDuration `yaml:"period"`
+	Timeout  int                `yaml:"timeout"`
+	Probe    string             `yaml:"type"`
```

Comment:
I'd keep this named `type` instead of `probe`. There are so many things named `probe` around here anyway (and I'm suggesting making more, via my first comment re s/observer/prober/ above!) that calling this "type" or "kind" would I think be clearer and more readable.

Reply:
Rewrote it to use type, and also just have the Prober provide it's own name rather than pass it around.